src/renderer/entities/transactionBuilder/lib/compound-wallet.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  CallBuilder,
+  TransactionBuilder,
+  TransactionVisitor
+} from "@entities/transactionBuilder/model/transaction-builder";
+import {AccountInWallet, Wallet} from "@shared/core/types/wallet";
+import {Account} from "@shared/core";
+import {Exception} from "@zxing/library";
+import {ApiPromise} from "@polkadot/api";
+import {BaseTxInfo, OptionsWithMeta, UnsignedTransaction} from "@substrate/txwrapper-polkadot";
+import {SubmittableExtrinsic} from "@polkadot/api/types";
+
+export class CompoundWalletTransactionBuilder implements TransactionBuilder {
+
+  effectiveCallBuilder: CallBuilder;
+
+  readonly wallet: Wallet
+  readonly shards: Account[]
+
+  readonly api: ApiPromise
+
+  #inner: TransactionBuilder
```

Comment:
What about `internalBuilder`?

Reply:
For me `internal` sounds a bit confusing, since it creates an association with `somethihg hidden from the outside`. Whereas this meaning is indeed true, imho it is not the main one - what we are really referring to here is that `inner` is the child tx builder that is wrapped by ` CompoundWalletTransactionBuilder`. What do you think about `nested` or `child` naming?

Reply:
Agreed on `innerBuilder`